src/serialization/tokenizer.cpp

```diff
@@ -68,84 +68,116 @@ const token &tokenizer::next_token()
 		next_char_fast();
 	}
 
+	// skip comments on their own line
 	if (current_ == token::POUND)
 		skip_comment();
 
+	// set the line number the next token will start on
 	startlineno_ = lineno_;
 
 	switch(current_) {
+	// we reached the end of the file being read
 	case EOF:
 		token_.type = token::END;
 		break;
 
+	// handle open/closed angle brackets
+	// most commonly used for enclosing lua code
+	// more generally is used to indicate the preprocessor should skip over a particular block of text
 	case token::LEFT_ANGLE_BRACKET:
+		// if there aren't double left angle brackets, there is no extra handling needed - this is just a regular left angle bracket
 		if (peek_char() != token::LEFT_ANGLE_BRACKET) {
 			token_.type = token::MISC;
 			token_.value += current_;
 			break;
 		}
+
+		// else, treat this like a quoted string
 		token_.type = token::QSTRING;
 		next_char_fast();
+
+		// keep getting characters and appending them to the current token's value until either the file ends or double right angle brackets are found
+		// finding the end of the file first is an error since double left angle brackets must always be closed by double right angle brackets
 		for (;;) {
 			next_char();
 			if (current_ == EOF) {
 				token_.type = token::UNTERMINATED_QSTRING;
 				break;
-			}
-			if (current_ == token::RIGHT_ANGLE_BRACKET && peek_char() == token::RIGHT_ANGLE_BRACKET) {
+			} else if (current_ == token::RIGHT_ANGLE_BRACKET && peek_char() == token::RIGHT_ANGLE_BRACKET) {
 				next_char_fast();
 				break;
 			}
 			token_.value += current_;
 		}
 		break;
 
+	// very similar to the double left+right angle bracket handling
+	// the main difference is the need to handle INLINED_PREPROCESS_DIRECTIVE_CHAR since double quotes don't affect the preprocessor
 	case token::DOUBLE_QUOTE:
 		token_.type = token::QSTRING;
+
 		for (;;) {
 			next_char();
 			if (current_ == EOF) {
 				token_.type = token::UNTERMINATED_QSTRING;
 				break;
+			} else if (current_ == token::DOUBLE_QUOTE) {
+				if (peek_char() != token::DOUBLE_QUOTE) {
+					break;
+				} else {
+					next_char_fast();
+				}
 			}
-			if (current_ == token::DOUBLE_QUOTE) {
-				if (peek_char() != token::DOUBLE_QUOTE) break;
-				next_char_fast();
-			}
+
+			// ignore this line and decrement the current line number
 			if (current_ == INLINED_PREPROCESS_DIRECTIVE_CHAR) {
 				skip_comment();
 				--lineno_;
 				continue;
 			}
+
 			token_.value += current_;
 		}
 		break;
 
+	// tag name delimiters
 	case token::OPEN_BRACKET:
 	case token::CLOSE_BRACKET:
+	// closing tag
 	case token::SLASH:
 	case token::NEWLINE:
 	case token::EQUALS:
+	// handles multiple attributes on the same line
+	// ie: x,y = 5,5
 	case token::COMMA:
+	// tag merge aka node append
```

Comment:
This isn't entirely correct, is it? The plus has two functions, depending on where it appears. It's string concatenation and line continuation if it appears to the right of an `=`.